fetch_data.py

```diff
@@ -162,6 +166,8 @@ def fetch_friendships(apis, users, excluded, out, target,
                         next_cursor, previous_cursor, new_user_friends,  = \
                             apis[api_idx].GetFriendIDsPaged(user_id=user["id"], stringify_ids=True, cursor=next_cursor)
                         user_friends = user_friends + new_user_friends
+                        if not user_friends:
+                            user_friends = [""]
```

Comment:
This check seems like it could be brittle. Does it instead make more sense to change the `while not user_friends:` check on https://github.com/eleurent/twitter-graph/pull/30/files#diff-afcd4836ccef17f66d84e9b8c2b38ed8fd9566a9e184bccc8e39dd759e402f2eR161? We could set a flag marking whether the API has successfully been called without a rate-limit error, and break out of the while loop if the flag has been set but `user_friends` is still empty.

Reply:
Your solution work for the "unstuck" part, but saving an empty edge for this specified user could be better to interpret the result at the end.

Just break out will skip the user and make no entry for him in the edges table. Like when we skip a private account.

Setting user_friends to [""] will save explicitly in the final result that this user has no edges.
This make a code maybe a little less pretty, but i want to keep a very explicit and non ambiguous result: 
no entry in edges table != an entry with no edges

But we may do both, refactoring this while and saving empty edges when we have to.

Reply:
+1 that the output dictionary should contain the user even if they have no outgoing edges.

Reply:
@FlorentLefebvre I see the distinction you are trying to make between no result, and the result that the list of friends is empty. However, in my opinion, an empty list should be represented by `[]` rather than a list containing a dummy id string `[""]`, which is not as clean and could have downstream effects (need to add additional checks that an id is valid, etc). And we could use the convention that `user_friends` is initialized or set to None when we have no result.
I have to admit that I do not understand the logic of the `while not user_friends` loop above, why is it needed?

Reply:
Ok, I think that I see now that this [""] convention was already there for handling errors below, and that only the second of the two nested while loop is needed: the first was originally there to retry after a rate limit error, and the second one was introduced for the paging mechanism, but it can also fulfill the rate limit retry mission and this will deal with the problem of users with no friends being stuck forever. I'll propose a fix